src/Components/Index/ServiceSelectionComponent.tsx

```diff
@@ -87,41 +83,23 @@ export class ServiceSelectionComponent extends SceneObjectBase<ServiceSelectionC
   }
 
   private _onActivate() {
-    // terribad hack - remove single service filter if it's there
-    const variable = sceneGraph.lookupVariable(VAR_FILTERS, this);
-    if (variable instanceof AdHocFiltersVariable && variable.state.filters.find((f) => f.key === SERVICE_NAME)) {
-      variable.setState({
-        filters: variable.state.filters.filter((f) => f.key !== SERVICE_NAME),
-      });
-    }
-
-    const unsubs: Unsubscribable[] = [];
-
-    const liveTailControl = getLiveTailControl(this);
-    if (liveTailControl) {
-      unsubs.push(
-        liveTailControl.subscribeToState(({ liveStreaming }) => {
-          Object.values(this._services).forEach((service) => {
-            service.logsQueryRunner.setState({
-              liveStreaming,
-              queries: [...service.logsQueryRunner.state.queries],
-            });
-            service.logsQueryRunner.runQueries();
-          });
-        })
-      );
-    }
-
-    this._onTopServiceChange();
-
+    this._getListOfServicesOrderedByVolume();
     this.subscribeToState((newState, oldState) => {
-      if (newState.topServicesToBeUsed !== oldState.topServicesToBeUsed) {
-        this.updateBody();
+      // Updates topServicesToBeUsed when listOfServicesOrderedByVolume is changed - should happen only once when the list of services is fetched during initialization
+      if (newState.listOfServicesOrderedByVolume !== oldState.listOfServicesOrderedByVolume) {
+        const ds = sceneGraph.lookupVariable(VAR_DATASOURCE, this)?.getValue();
+        const topServicesToBeUsed = addFavoriteServices(
+          newState.listOfServicesOrderedByVolume?.slice(0, LIMIT_SERVICES) ?? [],
+          getFavoriteServicesFromStorage(ds)
+        );
+        this.setState({
+          topServicesToBeUsed,
+        });
```

Comment:
Instead of having this logic in `_getListOfServicesOrderedByVolume` we are going to update `listOfServicesToQuery` when `listOfServicesOrderedByVolume` changes, which happens on initial load, or when user changes data source.